### PR TITLE
[4.0] Add prepared statements for plg_extension_joomla

### PR DESCRIPTION
**Pull request by HLeithner**
This PR has to be merged first https://github.com/joomla-framework/database/pull/156

### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
This plugin interacts with extension installation/removal/updates

* Install extensions
* Check for updates
* Update extensions
* Remove extensions

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/extension/joomla/joomla.php [hunk 1]
```diff
@@ -145,33 +166,38 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
 		if ($eid && $removed)
 		{
 			$db = Factory::getDbo();
-			$query = $db->getQuery(true)
-				->delete('#__update_sites_extensions')
-				->where('extension_id = ' . $eid);
+			$query = $db->getQuery(true);
+
+			$query->delete($db->quoteName('#__update_sites_extensions'))
+				->where($db->quoteName('extension_id'), ' = :eid')
```
Comment (Quy) [inline plugins/extension/joomla/joomla.php]:
```suggestion
				->where($db->quoteName('extension_id') . ' = :eid')
```

File: plugins/extension/joomla/joomla.php [hunk 2]
```diff
@@ -145,33 +166,38 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
+				->bind(':eid',  (int) $eid, ParameterType::INTEGER);
```
Comment (Quy) [inline plugins/extension/joomla/joomla.php]:
```suggestion
				->bind(':eid', (int) $eid, ParameterType::INTEGER);
```

File: plugins/extension/joomla/joomla.php [hunk 3]
```diff
@@ -85,20 +95,31 @@ private function addUpdateSite($name, $type, $location, $enabled)
...
+				->select($db->quoteName('update_site_id'))
+				->from($db->quoteName('#__update_sites_extensions'))
+				->where(
+					[
+						$db->quoteName('update_site_id') . ' = :updatesiteid',
+						$db->quoteName('extension_id') . ' = :extensionid'
+					]
+				)
+				->bind(':updatesiteid', $update_site_id, ParameterType::INTEGER)
+				->bind(':extensionid',  $this->eid, ParameterType::INTEGER);
```
Comment (Quy) [inline plugins/extension/joomla/joomla.php]:
```suggestion
				->bind(':extensionid', $this->eid, ParameterType::INTEGER);
```

**Comment by HLeithner** [comment]
Thanks @Quy you did an awesome job on all the PRs!

**Comment by HLeithner** [comment]
And thanks to @franz-wohlkoenig adding all the missing [4.0] to the title... my bad...

File: plugins/extension/joomla/joomla.php [hunk 4]
```diff
@@ -57,20 +58,29 @@ private function addUpdateSite($name, $type, $location, $enabled)
...
-				->insert('#__update_sites')
-				->columns(array($db->quoteName('name'), $db->quoteName('type'), $db->quoteName('location'), $db->quoteName('enabled')))
-				->values($db->quote($name) . ', ' . $db->quote($type) . ', ' . $db->quote($location) . ', ' . (int) $enabled);
+				->insert($db->quoteName('#__update_sites'))
+				->columns($db->quoteName(['name', 'type', 'location', 'enabled']))
+				->values(':name, :type, :location, :enabled')
+				->bind(':name', $name)
+				->bind(':type', $type)
+				->bind(':location', $location)
+				->bind(':enabled', (int) $enabled, ParameterType::INTEGER);
```
Comment (alikon) [inline plugins/extension/joomla/joomla.php]:
```suggestion
				->bind(':enabled', $enabled, ParameterType::INTEGER);
```

File: plugins/extension/joomla/joomla.php [hunk 5]
```diff
@@ -145,33 +166,38 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
 		{
 			$db = Factory::getDbo();
-			$query = $db->getQuery(true)
-				->delete('#__update_sites_extensions')
-				->where('extension_id = ' . $eid);
+			$query = $db->getQuery(true);
+
+			$query->delete($db->quoteName('#__update_sites_extensions'))
+				->where($db->quoteName('extension_id') . ' = :eid')
+				->bind(':eid', (int) $eid, ParameterType::INTEGER);
```
Comment (alikon) [inline plugins/extension/joomla/joomla.php]:
```suggestion
				->bind(':eid', $eid, ParameterType::INTEGER);
```

File: plugins/extension/joomla/joomla.php [hunk 6]
```diff
@@ -196,8 +223,10 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
 
 			// Last but not least we wipe out any pending updates for the extension
 			$query->clear()
-				->delete('#__updates')
-				->where('extension_id = ' . $eid);
+				->delete($db->quoteName('#__updates'))
+				->where($db->quoteName('extension_id') . ' = :eid')
+				->bind(':eid', (int) $eid, ParameterType::INTEGER);
```
Comment (alikon) [inline plugins/extension/joomla/joomla.php]:
```suggestion
				->bind(':eid', $eid, ParameterType::INTEGER);
```

**Review by alikon — APPROVED** [review]

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**